src/acq/channel.c

```diff
@@ -365,6 +365,8 @@ void bl_acq_channel_commit_sample(enum bl_acq_source source, uint32_t sample)
 		msg->sample_data.data32[count] = sample;
 
 		if (msg->sample_data.count >= MSG_SAMPLE_DATA32_MAX) {
+			msg->sample_data.src_led = bl_led_src[bl_led_active].led;
```

Comment:
What if the ADC had two enabled LEDs and we're alternating between them?

The `msg` comes from the channel (`union bl_msg_data *msg = channel->msg;`).

So the channel will be handling more than one LED, so this should probably be an LED mask?

Reply:
I realised this is problematic, maybe we have to separate the msg for each LED within the channel.